sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -39,6 +41,18 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
   ): Promise<InstitutionLocation> {
     const auditUser = { id: auditUserId } as User;
     const institution = { id: institutionId };
+
+    const isInstitutionCodeDuplicate = await this.hasLocationCodeForInstitution(
```

Comment:
I believe that this method is used during insert and update. During an update, the method `hasLocationCodeForInstitution` will ignore the duplication if the code is already assigned to itself, right?
The bottom line, the code allows inserting and updating properly, right?

Reply:
We don't use that method to update. `saveLocation` is only for the creation, there's another method the update. Then I didn't add any checks. The update method doesn't check for duplicates because it doesn't update the locationCode.

Reply:
As we talked, lets have this method renamed to `createLocation` and add the validation to the update method executed by AEST.